app/models/aspace/resource.rb

```diff
@@ -1,23 +1,40 @@
 # frozen_string_literal: true
 
 module Aspace
+  class AspaceResourceError < StandardError; end
+
   # Model for an Aspace resource to manage information
   # needed for downloading and updating resources in ArcLight
   class Resource
-    attr_reader :ead_id, :uri, :repository_code
+    attr_reader :ead_id, :uri, :identifier, :repository_code
 
-    def initialize(ead_id:, uri:, repository_code:)
+    def initialize(uri:, repository_code:, ead_id: nil, identifier: nil)
       @ead_id = ead_id
+      @identifier = identifier
       @uri = uri
       @repository_code = repository_code
     end
 
     def file_name
-      ead_id.sub(/(\.xml)?$/i, '').parameterize
+      resource_id.sub(/(\.xml)?$/i, '').parameterize
+    end
+
+    def resource_id
```

Comment:
`resource_id` could be confused with the aspace resource id. Some other ideas: `call_number`, `file_identifier`, `finding_aid_identifier`? Open to ideas.

Reply:
Should this be a private method?

Reply:
Good thinking. I like file_identifier I think?